src/containers/Desk.js

```diff
@@ -7,47 +7,72 @@ import Write from '../components/Write'
 class Desk extends Component {
   state = {
     letterStack: [],
-    logged_in: null
+    loggedIn: null,
+    intervalId: null,
+    plane: null
+  }
+
+  startPlanes = () => {
+    const planeInterval = setInterval(this.throwPlane, 5000)
+    this.setState({ intervalId: planeInterval })
+  }
+
+  stopPlanes = () => {
+    clearInterval(this.state.intervalId);
+  }
+
+  throwPlane = () => {
+    if(this.state.letterStack.length === 0) {
+      this.stopPlanes()
+      this.fetchLetters()
+      return;
+    }
+
+    const stack = [...this.state.letterStack]
+    const plane = stack.pop();
+    this.setState({ letterStack: stack, plane: plane })
   }
 
   componentDidMount() {
-    // Fetch the letters and setState using PrevState to update
-    this.fetchLetters();
+    this.fetchLetters()
+  }
+
+  componentWillUnmount() {
+    this.stopPlanes()
   }
 
   fetchLetters = () => {
-    // Call this function on an interval to get more letters
-    // Ideally grab 10-20 at a time? Do it every 30 seconds?
+    // TODO: grab the 10 most recent letters this account
+    // has not yet seen (make a new api on back end)
     const LETTERS_URL = 'http://localhost:3000/letters'
 
     fetch(LETTERS_URL)
       .then(resp => resp.json())
       .then(json => {
         this.setState({ letterStack: json })
+        this.startPlanes()
       })
   }
 
   renderWrite = () => {
-    console.log('write')
     return <Write />
   }
 
-  renderRead = e => {
-    console.log(e.target.id, this)
-    return <Read />
+  renderRead = letter => {
+    // TODO: mark the letter as seen for the given account
+    return <Read letter={letter} />
   }
 
-  renderPlanes = () => {
-    return this.state.letterStack.map(letter => {
-      return <Plane key={letter.id} letter={letter} handleClick={this.renderRead} />
-    })
+  renderPlane = () => {
+    if (this.state.plane)
+      return <Plane key={this.state.plane.id} plane={this.state.plane} handleClick={this.renderRead} />
```

Comment:
for handleClick - i made a handlePlaneClick function that will setState isRead: true, and then call renderRead with the plane/letter object

Reply:
Perfect! Sounds like it should be an easy merge